src/cephadm/cephadm

```diff
@@ -4623,7 +4627,10 @@ def _get_parser():
     parser_bootstrap.add_argument(
         '--apply-spec',
         help='Apply cluster spec after bootstrap (copy ssh key, add hosts and apply services)')
-
+    parser_bootstrap.add_argument(
+        '--ipv6',
+        action='store_true',
+        help='Tell the services to bind to ipv6')
```

Comment:
> Another approach would be to detect the address family from the
> `--mon-ip` and set appropriately. But that may open up having to dns
> resolve hostnames etc.

sadly the `ipaddress` module doesn't appear to always be present in py2.

but maybe `attempt_bind` can be modified to detect IPv4/IPv6 ?
```
>>> s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
>>> s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
>>> s.bind(('::', 1234))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.gaierror: [Errno -9] Address family for hostname not supported


>>> s = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
>>> s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
>>> s.bind(('0.0.0.0', 1234))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 99] Cannot assign requested address
```

In either case it might be wise to retain the new `--ipv6` argument as a failsafe ... ?

Reply:
yeah, it would kinda be better if we could have a full dual stack solution too, where services would bind to both ipv4 and 6. But not sure how well that could work or what ramifications that would mean to the messenger code.

I think it might be time for me to fall down that rabbit hole, just out of interest sake. As I think the person bootstrapping should know if they want ipv4 or 6, so --ipv6 is the easy path :)

Reply:
maybe we can deduce iPv6 by looking at the mon-ip? https://docs.ceph.com/docs/master/cephadm/install/#bootstrap-a-new-cluster